### PR TITLE
Updates to the base for hacky

### PR DESCRIPTION
**Pull request by treyhyde**
I changed the API which will make merge back painful.

I replaced "baseEntity" which is meaningless to me to pass in the `koa` context which we need to grab incoming headers.  I probably should have just added ctx

### CODE REVIEW

File: lib/scimgateway.js [hunk 1]
```diff
@@ -224,11 +225,12 @@ const ScimGateway = function () {
...
-    logger.error(`${gwName}[${pluginName}] Scimgateway password decryption failed or no password defined`)
-    logger.error(`${gwName}[${pluginName}] stopping...\n`)
-    throw (new Error('Using exception to stop further asynchronous code execution (ensure synchronous logger flush to logfile and exit program), please ignore this one...'))
-  }
+  // if (!foundBasic && !foundBearerToken && !foundBearerJwtAzure && !foundBearerJwt) {
+  //   logger.error(`${gwName}[${pluginName}] Scimgateway password decryption failed or no password defined`)
+  //   logger.error(`${gwName}[${pluginName}] stopping...\n`)
+  //   throw (new Error('Using exception to stop further asynchronous code execution (ensure synchronous logger flush to logfile and exit program), please ignore this one...'))
+  // }
+
```
Comment (treyhyde) [inline lib/scimgateway.js:233]: I think we can restore this now

File: lib/scimgateway.js [hunk 2]
```diff
@@ -426,6 +428,10 @@ const ScimGateway = function () {
     throw new Error('JWT authentication failed')
   }
 
+  const authNone = (authType) => {
+    return Promise.resolve(true)
+  }
+
```
Comment (treyhyde) [inline lib/scimgateway.js:434]: I don't think I need this anymore

File: lib/scimgateway.js [hunk 3]
```diff
@@ -489,15 +495,19 @@ const ScimGateway = function () {
         bearerToken(baseEntity, ctx.request.method, authType, authToken),
         bearerJwtAzure(baseEntity, ctx, next, authType, authToken),
         bearerJwt(baseEntity, ctx.request.method, authType, authToken),
-        bearerOAuth(baseEntity, ctx.request.method, authType, authToken)])
+        bearerOAuth(baseEntity, ctx.request.method, authType, authToken),
+        authNone(authType)
```
Comment (treyhyde) [inline lib/scimgateway.js:499]: I think we can drop it

File: lib/scimgateway.js [hunk 4]
```diff
@@ -489,15 +495,19 @@ const ScimGateway = function () {
...
+      ])
         .catch((err) => { throw (err) })
       for (const i in arrResolve) {
+        logger.info("method " + arrResolve[i]);
```
Comment (treyhyde) [inline lib/scimgateway.js:503]: remove

File: lib/scimgateway.js [hunk 5]
```diff
@@ -512,6 +522,7 @@ const ScimGateway = function () {
       ctx.body = { error: 'Access denied' }
       if (ctx.url !== '/favicon.ico') logger.error(`${gwName}[${pluginName}] ${err.message}`)
     } catch (err) {
+      logger.info("caught error, trying dumb methods " + err);
```
Comment (treyhyde) [inline lib/scimgateway.js:525]: remove

**Comment by treyhyde** [comment]
probably don't want to merge  in main.  Some changes need to be "fixed" and contributed back, some probably need to be a permanent fork.

**Comment by treyhyde** [comment]
https://github.com/fulcrumapp/fulcrum-js/pull/34
https://github.com/fulcrumapp/fulcrum/pull/6670
https://github.com/fulcrumapp/scimgateway/pull/1
https://github.com/fulcrumapp/scim-plugin-fulcrum/pull/1

**Comment by chrisworkmode** [comment]
This is being handled by
https://github.com/jelhub/scimgateway/pull/77
https://github.com/fulcrumapp/scim/pull/23
Closing this one.

**Pull request closed without merging**